aten/src/ATen/native/quantized/cpu/qnnpack/src/q8gemm_sparse/8x4c1x4-dq-sse2.c

```diff
@@ -0,0 +1,442 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <immintrin.h>
+
+#include <qnnpack/q8gemm_sparse.h>
+#include <requantization/runtime-sse2.h>
+
+#define MR 8
+#define COL_BLOCK_SIZE 4
+#define PACKED_A_BLOCK_SIZE COL_BLOCK_SIZE*MR
+
+#define LOAD_TRANSPOSED_A_BLOCK(id, index, col_block_id)                  \
```

Comment:
I personally lean on using forced inline functions if possible.  The thing with macros like this is that if you end up accidentally using any arithmetic in the arguments at call-site LOAD_TRANSPOSED_A_BLOCK(a, b, c+1) this will have unexpected consequences.  Also the commonality like `col_block_id * COL_BLOCK_SIZE + index` can be factored out more readily when using functions.

Reply:
I opted this way mainly because I can generate variables names like va0, va1 etc. that are later used and can avoid passing 10 args to the function. I agree about arithmetics used in arg leading to unexpected consequences. I will fix it appropriately.